Ginger/GingerCoreNET/AnalyzerLib/AnalyzeAction.cs

```diff
@@ -525,6 +493,72 @@ private static void GetAllStoreToGlobalParameters(Act a, ObservableList<GlobalAp
             }
         }
 
+        public static void AnalyzeValueExpInAction(Act action, BusinessFlow businessFlow, Activity activity, ref List<AnalyzerItemBase> issues)
+        {
+
+
+            var ValueExpsNotInCurrEnv = action.ActInputValues
+                                .Where((actInputValue) =>
+                                !AnalyzeEnvApplication.DoesEnvParamOrURLExistInValueExp(actInputValue.Value, businessFlow.Environment)
+                                ).Select((filteredActInputVal) => "Operation Settings")
+                                .ToList();
+
+
+            if (!AnalyzeEnvApplication.DoesEnvParamOrURLExistInValueExp(action.RunDescription, businessFlow.Environment)){
+
+                ValueExpsNotInCurrEnv.Add("Run Description");
+            }
+
+
+            var FlowControlValues = action
+                                    .ActFlowControls
+                                    .Where((actFlowControl) =>
+                                    {
+                                        return !AnalyzeEnvApplication.DoesEnvParamOrURLExistInValueExp(actFlowControl.Condition, businessFlow.Environment) ||
+                                        !AnalyzeEnvApplication.DoesEnvParamOrURLExistInValueExp(actFlowControl.Value, businessFlow.Environment);
+                                    })
+                                    .Select((filteredFlowControl) => "Flow Control");
+
+
+
+            var ReturnValues = action
+                                .ActReturnValues
+                                .Where((actReturnValue) =>
+                                {
+                                    return
+                                    !AnalyzeEnvApplication.DoesEnvParamOrURLExistInValueExp(actReturnValue.Param, businessFlow.Environment) ||
+                                    !AnalyzeEnvApplication.DoesEnvParamOrURLExistInValueExp(actReturnValue.Path, businessFlow.Environment) ||
+                                    !AnalyzeEnvApplication.DoesEnvParamOrURLExistInValueExp(actReturnValue.Expected, businessFlow.Environment);
+                                })
+                                .Select((filteredReturnValue) => "Output Values");
+
+            ValueExpsNotInCurrEnv.AddRange(FlowControlValues);
+            ValueExpsNotInCurrEnv.AddRange(ReturnValues);
+
+
+
+
+            foreach (var filteredValueExp in ValueExpsNotInCurrEnv)
+            {
+                AnalyzeAction AA = new AnalyzeAction();
+                AA.Status = eStatus.NeedFix;
+                AA.mActivity = activity;
+                AA.Description = $"Cannot Calculate Value Expression used in {filteredValueExp} of the {action.Description} ";
```

Comment:
Please also add what is missing, e.g. param or Url, also the problematic value expression